pandas/core/indexes/datetimes.py

```diff
@@ -315,16 +316,18 @@ def _simple_new(cls, values, name=None, freq=None, tz=None, dtype=None):
         if we are passed a non-dtype compat, then coerce using the constructor
         """
         # DatetimeArray._simple_new will accept either i8 or M8[ns] dtypes
-        values = DatetimeArray._simple_new(values, freq=freq, tz=tz)
-
+        if isinstance(values, DatetimeIndex):
+            values = values._data
```

Comment:
not sure if this is necessary, was part of troubleshooting

Reply:
it would also be really weird if `values` were passed as a DTA but values.freq didn't match the `freq` kwarg